openedx/features/content_tagging/rest_api/v1/filters.py

```diff
@@ -0,0 +1,45 @@
+"""
+API Filters for content tagging org
+"""
+
+from django.db.models import Exists, OuterRef, Q
+from rest_framework.filters import BaseFilterBackend
+
+from ...models import TaxonomyOrg
+from ...rules import get_user_taxonomy_orgs, is_global_taxonomy_admin
+
+
+class UserOrgFilterBackend(BaseFilterBackend):
+    """
+    Filter taxonomies based on the user's organization.
+    """
+
+    def filter_queryset(self, request, queryset, view):
+        if is_global_taxonomy_admin(request.user):
+            return queryset
+        else:
+            user_orgs = get_user_taxonomy_orgs(request.user)
+            user_orgs_short_name = [org.short_name for org in user_orgs]
```

Comment:
Hmm.. since `get_user_taxonomy_orgs` is only used here, and half of that method's logic is in the lines above with your `is_global_taxonomy_admin` check.. I think this should be refactored.

Maybe replace `rules.get_user_taxonomy_orgs` with this, so we're still isolating the `is_content_creator` check inside the rules:

```
def is_taxonomy_admin_for_org(user: User, org: Organization) -> bool:
    return is_content_creator(user, org.short_name)
```

And do your org filtering here:
```suggestion
            user_orgs_short_name = [
                org.short_name if is_taxonomy_admin_for_org(user, org)
                for org in Organization.objects.all()
            ]
```

Reply:
Done 98cb2eb68472c30b201988a2e5aa7a041104da98